File: src/main/java/nl/wernerkroneman/Drawy/Modelling/PlaceholderModel.java
package nl.wernerkroneman.Drawy.Modelling;

/**
 * Class representing a placeholder in a model,
 * something "unresolved".
 *
 * Usually, this model and any model containing it
 * cannot be represented directly.
 *
 * "object" is usually a PlaceholderModel.
 */
public class PlaceholderModel extends Model {

    public PlaceholderModel(String name) {
        super(name);
    }

    @Override
    public <V> V accept(ModelVisitor<V> visitor) {
        return visitor.visit(this);
    }
}

File: src/main/java/nl/wernerkroneman/Drawy/Modelling/PlaceholderModel.kt
/*
 * Copyright (c) 2017 Werner Kroneman
 *
 * This file is part of DrawyMcDrawface.
 *
 * DrawyMcDrawface is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * DrawyMcDrawface is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with DrawyMcDrawface.  If not, see <http://www.gnu.org/licenses/>.
 */

package nl.wernerkroneman.Drawy.Modelling

/**
 * Class representing a placeholder in a model,
 * something "unresolved".

 * Usually, this model and any model containing it
 * cannot be represented directly.

 * "object" is usually a PlaceholderModel.
 */
class PlaceholderModel(name: String) : Model(name) {

    override fun <V : Any> accept(visitor: ModelVisitor<V>): V {
        return visitor.visit(this)
    }
}